src/pks/flow/constitutive_relations/thaw_depth/gas_content_evaluator.cc

```diff
@@ -17,19 +17,20 @@ namespace Flow {
 GasContentEvaluator::GasContentEvaluator(Teuchos::ParameterList& plist)
     : SecondaryVariableFieldEvaluator(plist)
 {
-  domain_ = Keys::getDomain(my_key_);
-  auto pos = domain_.find_last_of('_');
-  int col_id = std::stoi(domain_.substr(pos+1, domain_.size()));
+  Key dset_name = plist.get<std::string>("domain set name", "column");
   
-  std::stringstream domain_ss;
-  domain_ss << "column_"<< col_id;
-  temp_key_ = Keys::getKey(domain_ss.str(),"temperature");
+  domain_ = Keys::getDomain(my_key_); //surface_column domain
+  int col_id = Keys::getDomainSetIndex<int>(domain_);
+  
+  Key domain_ss = Keys::getDomainInSet(dset_name, col_id);
+  
+  temp_key_ = Keys::getKey(domain_ss,"temperature");
```

Comment:
Can these change to Keys::readKey() calls?  Apparently there are still a lot of hard-coded names in the code, but we can fix them as we see them!

Reply:
done